### PR TITLE
feat(jtk): complete presenter boundary architecture

### PR DESCRIPTION
**Pull request by rianjs**
## Summary

- Eliminates generic presenter methods (`PresentSuccess/Info/Advisory/Warning/Error`) from command code — commands no longer choose wording or message kind
- Eliminates multi-fragment output orchestration — commands pass domain data to a single presenter method that returns a complete `OutputModel`
- Adds 11 domain-specific presenter methods covering move operations, pagination, field options, config show, and automation updates
- Removes 7 generic methods from `issue.go` and `field.go` after all usages are migrated

Replaces #225 (merged prematurely, reverted in 36291b5).

## Test plan

- [ ] `make build && make test && make lint` all pass
- [ ] `jtk config show` shows table + path info (both on stdout)
- [ ] `jtk issues move` with invalid type shows error + available types
- [ ] `jtk automation update` shows progress then success message

[MON-221]

### CODE REVIEW

**Comment by rianjs** [comment]
# Plan: Eliminate Presenter DTO Construction in Commands

**Issue:** #221 (PR #226 follow-up)

---

## Context

PR #226 eliminated generic presenter methods and multi-fragment orchestration. However, a third category of architectural violation remains: **commands construct presenter-owned DTOs** instead of passing raw domain/API data.

The violations:
- Commands import presenter DTO types (`jtkpresent.FieldContext`, `jtkpresent.ConfigEntry`, etc.)
- Commands build these DTOs via field-by-field mapping from API types
- Some commands embed presentation logic (e.g., appending "(disabled)" suffix)

**Target state:** Commands pass raw domain/API types to presenter methods. Presenters own all mapping from domain → presentation. Presenter DTOs become unexported or are removed entirely.

---

## Violations (7 locations)

### Category 1: Field-related DTOs

| Location | Command Builds | Source API Type |
|----------|---------------|-----------------|
| `fields/contexts.go:72` | `[]jtkpresent.FieldContext` | `[]api.FieldContext` |
| `fields/options.go:101` | `[]jtkpresent.FieldContextOption` | `[]api.FieldContextOption` |
| `issues/fields.go:75` | `[]jtkpresent.EditableField` | `map[string]any` (raw JSON) |
| `issues/field_options.go:102` | `[]jtkpresent.FieldOption` | `[]api.FieldOptionValue` |

### Category 2: Config DTOs

| Location | Command Builds | Source Data |
|----------|---------------|-------------|
| `configcmd.go:78` | `[]jtkpresent.ConfigEntry` | Raw config values + source helpers |
| `configcmd.go:270` | `jtkpresent.TestResult` | url, clientErr, user, authErr |

### Category 3: Move failure DTO

| Location | Command Builds | Source API Type |
|----------|---------------|-----------------|
| `move.go:180` | `[]jtkpresent.MoveFailedIssue` | `[]api.MoveFailedIssue` |

---

## Implementation Plan

### Phase 1: Field Contexts — Accept `[]api.FieldContext`

**Presenter change** (`field.go`):
```go
// Before: func (FieldPresenter) PresentContexts(contexts []FieldContext)
// After:
func (FieldPresenter) PresentContexts(contexts []api.FieldContext) *present.OutputModel
```

**Command change** (`fields/contexts.go`):
- Remove conversion loop (lines 72-80)
- Pass `result.Values` directly

**Cleanup**: Remove `FieldContext` struct from field.go

### Phase 2: Field Context Options — Accept `[]api.FieldContextOption`

**Presenter change** (`field.go`):
```go
// Before: func (FieldPresenter) PresentContextOptions(options []FieldContextOption)
// After:
func (FieldPresenter) PresentContextOptions(options []api.FieldContextOption) *present.OutputModel
```

**Command change** (`fields/options.go`):
- Remove conversion loop (lines 101-108)
- Pass `result.Values` directly

**Cleanup**: Remove `FieldContextOption` struct from field.go

### Phase 3: Move Failures — Accept `[]api.MoveFailedIssue`

**Presenter change** (`issue.go`):
```go
// Before: func (IssuePresenter) PresentMovePartialFailure(successful []string, failed []MoveFailedIssue)
// After:
func (IssuePresenter) PresentMovePartialFailure(successful []string, failed []api.MoveFailedIssue) *present.OutputModel
```

**Command change** (`issues/move.go`):
- Remove conversion loop (lines 180-183)
- Pass `status.Result.Failed` directly

**Cleanup**: Remove `MoveFailedIssue` struct from issue.go

### Phase 4: Field Options — Accept `[]api.FieldOptionValue`

**Presenter change** (`field.go`):
```go
// Before: func (FieldPresenter) PresentFieldOptionsWithHeader(fieldName string, options []FieldOption)
// After:
func (FieldPresenter) PresentFieldOptionsWithHeader(fieldName string, options []api.FieldOptionValue) *present.OutputModel
```

Move presentation logic into presenter:
- Fallback `Value` → `Name` when Value is empty
- Append "(disabled)" suffix

**Command change** (`issues/field_options.go`):
- Remove conversion loop (lines 102-115)
- Pass `options` directly

**Cleanup**: Remove `FieldOption` struct from field.go

### Phase 5: Editable Fields — Create API type, accept it

**API addition** (`tools/jtk/api/issues.go`):
```go
// EditFieldMeta represents field metadata from issue edit metadata API
type EditFieldMeta struct {
    ID       string
    Name     string
    Type     string // from schema.type
    Required bool
}

// ParseEditMeta extracts field metadata from raw edit metadata response
func ParseEditMeta(meta map[string]any) []EditFieldMeta
```

**Presenter change** (`field.go`):
```go
// Before: func (FieldPresenter) PresentEditableFields(fields []EditableField)
// After:
func (FieldPresenter) PresentEditableFields(fields []api.EditFieldMeta) *present.OutputModel
```

**Command change** (`issues/fields.go`):
- Replace manual extraction (lines 75-101) with `api.ParseEditMeta(fieldsData)`
- Pass result directly to presenter

**Cleanup**: Remove `EditableField` struct from field.go

### Phase 6: Config Show — Accept raw values with sources

The presenter must remain pure (no I/O, no env/config inspection). Source detection stays in the domain layer as a projection helper — this is not a quasi-presenter layer, just domain-side logic that knows how to inspect env vars and config files to determine where values came from.

**Config package addition** (`tools/jtk/internal/config/source.go`):
```go
// ConfigWithSources returns all config values with their sources
type ConfigWithSources struct {
    URL            string
    URLSource      string
    Email          string
    EmailSource    string
    APIToken       string  // unmasked - command will mask before passing to presenter
    TokenSource    string
    DefaultProject string
    ProjectSource  string
    AuthMethod     string
    AuthMethodSrc  string
    CloudID        string
    CloudIDSrc     string
    Path           string
}

func GetConfigWithSources() ConfigWithSources
```

Move source detection helpers from configcmd.go to `internal/config/`:
- `getURLSource()` → part of `GetConfigWithSources()`
- `getEmailSource()` → part of `GetConfigWithSources()`
- `getAPITokenSource()` → part of `GetConfigWithSources()`
- `getDefaultProjectSource()` → part of `GetConfigWithSources()`

**Presenter change** (`config.go`):
```go
// Presenter accepts pre-computed (value, source) pairs - no I/O
func (ConfigPresenter) PresentConfigShow(
    url, urlSrc, email, emailSrc, maskedToken, tokenSrc,
    defaultProject, projectSrc, authMethod, authMethodSrc,
    cloudID, cloudIDSrc, configPath string,
) *present.OutputModel
```

**Command change** (`configcmd.go`):
- Call `config.GetConfigWithSources()` to get all values + sources
- Mask token with `maskToken()`
- Pass all (value, source) pairs to presenter

**Key principle**: ConfigPresenter only formats what it is handed. All env/config inspection happens in `config.GetConfigWithSources()`.

**Cleanup**: Make `ConfigEntry` unexported → `configEntry`

### Phase 7: Config Test — Accept raw values as parameters

**Presenter change** (`config.go`):
```go
// Before: func (ConfigPresenter) PresentTestResult(r TestResult)
// After:
func (ConfigPresenter) PresentTestResult(url string, user *api.User, clientErr, authErr error) *present.OutputModel
```

**Command change** (`configcmd.go`):
- Remove `TestResult` struct construction
- Pass individual values

**Cleanup**: Remove `TestResult` struct from config.go

---

## Implementation Sequence

Order by complexity (simplest first):

1. Phase 1: Field Contexts (1:1 mapping, drops one field)
2. Phase 2: Field Context Options (1:1 mapping)
3. Phase 3: Move Failures (1:1 mapping, field rename)
4. Phase 4: Field Options (moves presentation logic)
5. Phase 5: Editable Fields (requires new API type)
6. Phase 6: Config Show (moves source helpers)
7. Phase 7: Config Test (parameter refactor)

**Execution strategy**: Phases 1-4 are mechanical and can be done by haiku subagents in parallel. Phases 5-7 require more judgment (creating API type, moving helpers).

---

## Verification

```bash
# Build and test
make build && make test && make lint

# ZERO presenter DTO construction in commands
grep -rn "jtkpresent\.\(FieldContext\|FieldContextOption\|EditableField\|FieldOption\|ConfigEntry\|TestResult\|MoveFailedIssue\)" tools/jtk/internal/cmd/ --include="*.go" | grep -v "_test.go"
# Expected: no output

# Presenter DTOs are unexported or removed
grep -rn "^type \(FieldContext\|FieldContextOption\|EditableField\|FieldOption\|ConfigEntry\|TestResult\|MoveFailedIssue\) struct" tools/jtk/internal/present/ --include="*.go"
# Expected: no output (all removed or lowercased)
```

---

## Critical Files

**Presenters:**
- `tools/jtk/internal/present/field.go`
- `tools/jtk/internal/present/config.go`
- `tools/jtk/internal/present/issue.go`

**Commands:**
- `tools/jtk/internal/cmd/fields/contexts.go`
- `tools/jtk/internal/cmd/fields/options.go`
- `tools/jtk/internal/cmd/issues/fields.go`
- `tools/jtk/internal/cmd/issues/field_options.go`
- `tools/jtk/internal/cmd/issues/move.go`
- `tools/jtk/internal/cmd/configcmd/configcmd.go`

**API (new):**
- `tools/jtk/api/issues.go` (add `EditFieldMeta` + `ParseEditMeta`)

**Config (new):**
- `tools/jtk/internal/config/source.go` (add `ConfigWithSources` + `GetConfigWithSources`)

**Comment by rianjs** [comment]
## TDD Assessment

### Untested presenter files (no test file exists)

The following new presenter files ship with **zero tests**:

| File | Public methods |
|------|---------------|
| `tools/jtk/internal/present/attachment.go` | `PresentList`, `PresentEmpty`, `PresentDeleted`, `PresentUploaded` |
| `tools/jtk/internal/present/board.go` | `PresentList`, `PresentDetail`, `PresentEmpty` |
| `tools/jtk/internal/present/comment.go` | `PresentList`, `PresentEmpty`, `PresentAdded`, `PresentDeleted` |
| `tools/jtk/internal/present/config.go` | `PresentTestResult`, `PresentConfigShow`, `PresentCleared`, `PresentClearedWithEnvVars`, `PresentNoConfig`, `PresentClearCancelled` |
| `tools/jtk/internal/present/dashboard.go` | `PresentList`, `PresentDetail`, `PresentEmpty`, `PresentCreated`, `PresentDeleted` |
| `tools/jtk/internal/present/field.go` | 14 public methods across all field/context/option CRUD |
| `tools/jtk/internal/present/helpers.go` | `FormatSize`, `TruncateText`, `FormatDate`, `FormatDateTime`, `BoolString`, `OrDash`, `FormatAssignee`, `FormatTime` |
| `tools/jtk/internal/present/link.go` | `PresentList`, `PresentEmpty`, `PresentAdded`, `PresentDeleted` |
| `tools/jtk/internal/present/project.go` | `PresentList`, `PresentDetail`, `PresentTypes`, and mutation methods |
| `tools/jtk/internal/present/transition.go` | `PresentList`, `PresentEmpty`, `PresentDone`, `PresentInvalidTransition` |

### Untested new API function

`api.ParseEditMeta` in `tools/jtk/api/issues.go` has no tests. This is a non-trivial parsing function that iterates over `map[string]any` with type assertions and fallback logic. Missing cases:

- Input with non-map values (the `!ok` skip path)
- Fields where `required` is absent vs. explicitly `false`
- Fields where `schema` key is absent (empty `fieldType`)
- Empty map input
- Fields with non-string name values hitting the `fmt.Sprintf("%v")` fallback in `safeString`

### Untested new config functions

`config.GetValuesWithSources` in `tools/jtk/internal/config/source.go` and all six `Get*WithSource` helpers have no tests. These functions inspect env vars in precedence order and fall back to the config file. Missing cases:

- Each env var precedence path (e.g., `JIRA_URL` wins over `ATLASSIAN_URL`)
- Config file fallback when no env var is set
- Legacy `JIRA_DOMAIN` / `cfg.Domain` paths in `GetURLWithSource`
- `Load()` error path (returns `"-"` as source)

### Gaps in tested presenters

**`config.PresentTestResult`** has no test file at all. This is the most branchy method in the PR (4 distinct code paths: no URL, client error, auth error, success) and is directly called out in the PR test plan.

**`issue.PresentMoveStatus`** is not directly tested. No test covers:
- `status.Result` with both `Successful` and `Failed` populated simultaneously
- `status.Result == nil` (only detail section, no appended messages)
- Multiple failed issues (the inner loop)

**`issue.PresentDetail` truncation** — `noTruncate=false` with a description longer than 200 chars is not tested. The `noTruncate=true` path is also absent. Both are dead code from a test perspective.

**`helpers.TruncateText`** — used by multiple presenters but has no test. The `maxLen <= 3` edge case and exact-length input are untested.

**`field.PresentFieldOptions`** — the `opt.Value == ""` fallback to `opt.Name` is untested, as is the `opt.Disabled` suffix path. Same gap in `PresentFieldOptionsWithHeader`.

File: tools/jtk/internal/cmd/automation/update.go [hunk 1]
```diff
@@ -72,12 +73,13 @@ func runUpdate(ctx context.Context, opts *root.Options, ruleID, filePath string)
 		return fmt.Errorf("fetching current rule: %w", err)
 	}
 
-	v.Info("Updating rule: %s (UUID: %s, State: %s)", current.Name, current.Identifier(), current.State)
-
 	if err := client.UpdateAutomationRule(ctx, ruleID, json.RawMessage(data)); err != nil {
```
Comment (monit-reviewer) [inline tools/jtk/internal/cmd/automation/update.go:76]: 🔵 **Low** (harness-engineering:enforcement-reviewer): Progress message timing changed: The old code emitted 'Updating rule...' BEFORE the API call, providing feedback during slow network operations. The new PresentUpdateComplete() is called AFTER the API call completes, so users have no 'in progress' indication while waiting. The progress message IS present in the output, just delayed. This is a minor UX regression for slow operations.

File: tools/jtk/internal/cmd/automation/enable.go [hunk 2]
```diff
@@ -44,14 +46,20 @@ func runSetState(ctx context.Context, opts *root.Options, ruleID string, enabled
 	}
 
 	if current.State == newState {
-		v.Info("Rule %q is already %s", current.Name, newState)
+		model := jtkpresent.AutomationPresenter{}.PresentNoChange(current.Name, newState)
```
Comment (monit-reviewer) [inline tools/jtk/internal/cmd/automation/enable.go:49]: 🔵 **Low** (harness-engineering:knowledge-reviewer): Behavioral change: 'already ENABLED/DISABLED' no-change messages moved from stdout to stderr. This is intentional (no mutation = advisory) and tests were updated, but any scripts capturing stdout to detect no-change state will silently stop seeing these messages. Consider documenting this in release notes.

File: tools/jtk/internal/cmd/dashboards/dashboards.go [hunk 3]
```diff
@@ -63,47 +68,47 @@ func newListCmd(opts *root.Options) *cobra.Command {
...
 
-	result, err := client.GetDashboards(0, maxResults)
-	if err != nil {
-		return err
-	}
-
-	if len(result.Dashboards) == 0 {
-		v.Info("No dashboards found")
+	if len(dashboards) == 0 {
+		model := jtkpresent.DashboardPresenter{}.PresentEmpty()
```
Comment (monit-reviewer) [inline tools/jtk/internal/cmd/dashboards/dashboards.go:96]: 🔵 **Low** (harness-engineering:knowledge-reviewer): Search context lost in empty-state message: When searching with --search flag returns no results, PresentEmpty() outputs generic 'No dashboards found' instead of 'No dashboards found matching "search term"'. Consider adding a PresentEmptySearch(term string) method.

File: shared/present/render.go [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+
+// escapeAgentCell escapes pipe characters and normalizes newlines in cell content.
+// This prevents cell content from being structurally indistinguishable from delimiters.
+func escapeAgentCell(s string) string {
+	s = strings.ReplaceAll(s, "\n", " ")
+	s = strings.ReplaceAll(s, "|", "\\|")
+	return s
+}
+
+func renderHumanTable(sec *TableSection) string {
```
Comment (monit-reviewer) [inline shared/present/render.go:95]: 🔵 **Low** (harness-engineering:architecture-reviewer): Bold headers missing in renderHumanTable: view.humanTable() applies bold formatting to headers, but renderHumanTable() does not. Commands migrated to the new presenter system lose bold header styling compared to unmigrated commands.

**Review by monit-reviewer — APPROVED** [review]
## Automated PR Review



**Reviewed commit:** `c7aa478`

> **Approved** with 5 non-blocking suggestions below. Address at your discretion.

### Summary

| Reviewer | Findings |
|----------|----------|
| harness-engineering:architecture-reviewer | 1 |
| harness-engineering:enforcement-reviewer | 1 |
| harness-engineering:knowledge-reviewer | 3 |

<details>
<summary><strong>harness-engineering:architecture-reviewer (1 findings)</strong></summary>

**💡 Suggestion** - `shared/present/render.go:95`
> Bold headers missing in renderHumanTable: view.humanTable() applies bold formatting to headers, but renderHumanTable() does not. Commands migrated to the new presenter system lose bold header styling compared to unmigrated commands.

</details>

<details>
<summary><strong>harness-engineering:enforcement-reviewer (1 findings)</strong></summary>

**💡 Suggestion** - `tools/jtk/internal/cmd/automation/update.go:76`
> Progress message timing changed: The old code emitted 'Updating rule...' BEFORE the API call, providing feedback during slow network operations. The new PresentUpdateComplete() is called AFTER the API call completes, so users have no 'in progress' indication while waiting. The progress message IS present in the output, just delayed. This is a minor UX regression for slow operations.

</details>

<details>
<summary><strong>harness-engineering:knowledge-reviewer (3 findings)</strong></summary>

**💡 Suggestion** - `tools/jtk/internal/cmd/automation/enable.go:49`
> Behavioral change: 'already ENABLED/DISABLED' no-change messages moved from stdout to stderr. This is intentional (no mutation = advisory) and tests were updated, but any scripts capturing stdout to detect no-change state will silently stop seeing these messages. Consider documenting this in release notes.

**💡 Suggestion** - `tools/jtk/internal/cmd/dashboards/dashboards.go:96`
> Search context lost in empty-state message: When searching with --search flag returns no results, PresentEmpty() outputs generic 'No dashboards found' instead of 'No dashboards found matching "search term"'. Consider adding a PresentEmptySearch(term string) method.

**💡 Suggestion** - `tools/jtk/internal/present:1`
> Missing unit tests for 9 presenter files: attachment.go, board.go, comment.go, config.go, dashboard.go, field.go, link.go, project.go, transition.go. These contain non-trivial formatting logic. The existing tests for automation, issue, mutation, sprint, and user set a good standard.

</details>

*3 info-level observations excluded. Run with `--verbose` to include.*

*1 PR discussion thread considered.*

---
<details>
<summary>Completed in 11m 21s | $3.99</summary>

| Field | Value |
|---|---|
| Reviewers | hybrid-synthesis, database:sql-reviewer, security:code-auditor, harness-engineering:knowledge-reviewer, harness-engineering:enforcement-reviewer, harness-engineering:architecture-reviewer, harness-engineering:legibility-reviewer |
| Engine | claude |
| Reviewed by | [pr-review-daemon](https://github.com/SignalFT/pr-review-daemon) · [monit-pr-reviewer](https://github.com/apps/monit-pr-reviewer) |
| Duration | 11m 21s wall · 21m 00s compute (Reviewers: 8m 17s · Synthesis: 3m 00s) |
| Cost | $3.99 |
| Tokens | 1561.3k in / 50.7k out |
| Turns | 39 |

</details>

*Note: Posted 4 inline comments. 1 finding remained in the summary because GitHub could not attach them to the current diff.*
